src/interfaces/noise_model.cpp

```diff
@@ -0,0 +1,19 @@
+#include "gl_depth_sim/interfaces/noise_model.h"
+
+std::vector<float> gl_depth_sim::noise(std::vector<float> distance)
+{
+  float segma;
+  std::random_device r;
+  std::seed_seq seed2{r(), r(), r(), r(), r(), r(), r(), r()};
+  std::mt19937 e2(seed2);
```

Comment:
A thought:

Re-creating these every function call is probably quite expensive. You might consider moving the `std::mt19937` object into a struct or something thats passed as an additional argument to the function. That would also let people control the seed sequence used to generate the noise. As it stands, they'll get a  totally new one for each operation. 

Don't worry about it for now, but if you don't understand the idea of random number seeds then please look it up.